test/test_tensorflow.py

```diff
@@ -631,35 +646,35 @@ def test_compression_fp16(self):
         tensor_size = [17] * 3
         compression = hvd.Compression.fp16
 
-        with self.test_session(config=self.config) as session:
-            for dtype in valid_dtypes:
-                tensor = tf.ones(tensor_size, dtype=dtype)
+        for dtype in valid_dtypes:
+            tensor = tf.ones(tensor_size, dtype=dtype)
 
-                tensor_compressed, ctx = compression.compress(tensor)
-                self.assertEqual(tensor_compressed.dtype, tf.float16)
+            tensor_compressed, ctx = compression.compress(tensor)
+            self.assertEqual(tensor_compressed.dtype, tf.float16)
 
-                tensor_decompressed = compression.decompress(tensor_compressed, ctx)
-                self.assertEqual(tensor_decompressed.dtype, dtype)
+            tensor_decompressed = compression.decompress(tensor_compressed, ctx)
+            self.assertEqual(tensor_decompressed.dtype, dtype)
 
-                actual = session.run(tensor_decompressed)
-                expected = np.ones(tensor_size)
-                err = np.linalg.norm(expected - actual)
-                self.assertLess(err, 0.00000001)
+            actual = self.evaluate(tensor_decompressed)
+            expected = np.ones(tensor_size)
+            err = np.linalg.norm(expected - actual)
+            self.assertLess(err, 0.00000001)
 
-            for dtype in invalid_dtypes:
-                tensor = tf.ones(tensor_size, dtype=dtype)
+        for dtype in invalid_dtypes:
+            tensor = tf.ones(tensor_size, dtype=dtype)
 
-                tensor_compressed, ctx = compression.compress(tensor)
-                self.assertEqual(tensor_compressed.dtype, dtype)
+            tensor_compressed, ctx = compression.compress(tensor)
+            self.assertEqual(tensor_compressed.dtype, dtype)
 
-                tensor_decompressed = compression.decompress(tensor_compressed, ctx)
-                self.assertEqual(tensor_decompressed.dtype, dtype)
+            tensor_decompressed = compression.decompress(tensor_compressed, ctx)
+            self.assertEqual(tensor_decompressed.dtype, dtype)
 
-                actual = session.run(tensor_decompressed)
-                expected = np.ones(tensor_size)
-                err = np.linalg.norm(expected - actual)
-                self.assertLess(err, 0.00000001)
+            actual = self.evaluate(tensor_decompressed)
+            expected = np.ones(tensor_size)
+            err = np.linalg.norm(expected - actual)
+            self.assertLess(err, 0.00000001)
 
 
 if __name__ == '__main__':
+    run_all_in_graph_and_eager_modes(MPITests)
```

Comment:
With this setting, is `self.config.gpu_options.allow_growth = True` still respected?